File: examples/source/CPP/ex189.cpp
#include "d4all.hpp"

#ifdef __TURBOC__
   extern unsigned _stklen = 10000;
#endif

void main()
{
   Code4 cb ;
   Data4 data( cb, "DATA1" ) ;
   Field4 field ;
   short j ;

   /* list the fields that are character fields */
   for (j = 1 ; j < data.numFields( ) ; j++ )
   {
      field.init( data, j ) ;
      if ( field.type( ) == r4str )
         cout << field.name( ) << endl ;
   }

   cb.initUndo( ) ;
}

File: examples/source/CSharp/ex189.cs
using System;
using CodeBase;
 
namespace c4ap
{
   class ex189
   {
      [STAThread]
      static void Main(string[] args)
      {
         Code4 cb = new Code4( ) ;
         Data4 data = new Data4( cb, "DATA1" ) ;
         short j ;
 
         /* list the fields that are character fields */
         for (j = 1 ; j < data.numFields( ) ; j++ )
         {
            Field4 field = new Field4( data, j ) ;
            if ( field.type( ) == Code4.r4str )
               Console.WriteLine( field.name( ) ) ;
         }
 
         cb.initUndo( ) ;
      }
   }
}